## 51_edit-env_copy-list.diff by Peter S Galbraith <psg@debian.org>

--- a/elisp/emacs-goodies-el/edit-env.el	2006-02-02 22:21:32.000000000 -0500
+++ b/elisp/emacs-goodies-el/edit-env.el	2006-02-02 22:22:38.000000000 -0500
@@ -111,6 +111,16 @@
 	       (list (widget-get widget 'environment-variable-name)
 		     widget)))
   
+;; Local copy from `copy-list' from cl.el  (PSG, Closes #340735)
+(defun edit-env-copy-list (list)
+  "Return a copy of a list, which may be a dotted list.
+The elements of the list are not copied, just the list structure itself."
+  (if (consp list)
+      (let ((res nil))
+	(while (consp list) (push (pop list) res))
+	(prog1 (nreverse res) (setcdr res list)))
+    (car list)))
+
 ;;;###autoload
 (defun edit-env ()
   "Display, edit, delete and add environment variables."
@@ -132,7 +142,7 @@
 	(val nil)
 	(longest-var 0)
 	(current-widget nil))
-    (setq edit-env-ls (copy-list process-environment))
+    (setq edit-env-ls (edit-env-copy-list process-environment))
     (setq edit-env-ls (sort edit-env-ls (lambda (a b) (string-lessp a b))))
 
     (widget-create 'push-button
